### PR TITLE
Updated Ada info

### PR DESCRIPTION
**Pull request by rkoeninger**
* Added extensions:
  * `.adc` - used for pragma-only files - [github search result](https://github.com/search?p=1&q=extension%3Aadc+pragma&type=Code&utf8=%E2%9C%93)
  * `.gpr` - GNAT project file that uses Ada syntax - [github search result](https://github.com/search?l=Ada&p=1&q=extension%3Agpr+project&type=Code&utf8=%E2%9C%93)
* Extensions put in alphabetical order
* Added example files under `samples/Ada` for `.adb`, `.adc`, `.ads`, `.gpr`, and `.ada`.
* Added `ada2012` alias
* Added `source.ada` as `tm_scope`

### CODE REVIEW

**Comment by pchaigno** [comment]
Please have a look at the [guidelines to add support for a new extension to Linguist](https://github.com/github/linguist/blob/master/CONTRIBUTING.md#adding-an-extension-to-a-language). In particular, notice the requirement on in-the-wild usage.

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -163,13 +163,17 @@ Ada:
   type: programming
   color: "#02f88c"
   extensions:
-  - ".adb"
```
Comment (pchaigno) [inline lib/linguist/languages.yml]: Why did you change this? Is `.adb` not the primary extension?

Comment (rkoeninger) [reply lib/linguist/languages.yml]: `.adb` is still in the list. Extensions were re-ordered to be alphabetical.

Comment (Alhadis) [reply lib/linguist/languages.yml]: @rkoeninger Note the part in @pchaigno's comment about **primary extension**. Linguist makes an exception to the alphabetisation rule in that the "primary" file extension (the one most commonly used or associated with a language) comes first, irrespective of anything else.

It's slightly confusing for new contributors, I'll admit. Please make sure you've read not just [`CONTRIBUTING.md`](https://github.com/github/linguist/blob/master/CONTRIBUTING.md), but also the field documentation at the top of the `languages.yml` file. You'll notice that the `extensions` field [has an explanation](https://github.com/github/linguist/blob/4cd558c/lib/linguist/languages.yml#L12-L14) about the "primary extension" matter:

> **`extensions`** - An Array of associated extensions (the first one is considered the primary extension, the others should be listed alphabetically)

Comment (rkoeninger) [reply lib/linguist/languages.yml]: Oh, sorry, I didn't see that. I'll push a correction.

**Comment by rkoeninger** [comment]
@pchaigno Does this PR need any more work or is it good to go?
